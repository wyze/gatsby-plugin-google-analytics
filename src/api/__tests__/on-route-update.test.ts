import onRouteUpdate from '../on-route-update'

declare module window {
  export let ga: jest.Mock<Function>
}

describe('onRouteUpdate', () => {
  it('calls ga function with new page', () => {
    process.env.NODE_ENV = 'production'
    window.ga = jest.fn()

    const location = { hash: '', pathname: '/about', search: '' }
    const actual = onRouteUpdate({ location })

    expect(window.ga).toHaveBeenCalled()
    expect(window.ga.mock.calls[0]).toMatchSnapshot('setting the page')
    expect(window.ga.mock.calls[1]).toMatchSnapshot('sending the pageview')

    process.env.NODE_ENV = 'test'
  })

  it('handles when location is not supplied', () => {
    process.env.NODE_ENV = 'production'
    window.ga = jest.fn()

    // const location = { hash: '', pathname: '/about', search: '' }
    const actual = onRouteUpdate()

    expect(window.ga).toHaveBeenCalled()
    expect(window.ga.mock.calls[0]).toMatchSnapshot('setting the page')
    expect(window.ga.mock.calls[1]).toMatchSnapshot('sending the pageview')

    process.env.NODE_ENV = 'test'
  })

  it('does nothing when not in production', () => {
    window.ga = jest.fn()

    const location = { hash: '', pathname: '/about', search: '' }
    const actual = onRouteUpdate({ location })

    expect(window.ga).not.toHaveBeenCalled()
  })
})
