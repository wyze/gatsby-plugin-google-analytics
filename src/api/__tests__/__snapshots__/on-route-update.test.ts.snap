// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onRouteUpdate calls ga function with new page: sending the pageview 1`] = `
Array [
  "send",
  "pageview",
]
`;

exports[`onRouteUpdate calls ga function with new page: setting the page 1`] = `
Array [
  "set",
  "page",
  "/about",
]
`;

exports[`onRouteUpdate handles when location is not supplied: sending the pageview 1`] = `
Array [
  "send",
  "pageview",
]
`;

exports[`onRouteUpdate handles when location is not supplied: setting the page 1`] = `
Array [
  "set",
  "page",
  undefined,
]
`;
